docs/data-layer.md

```diff
@@ -0,0 +1,178 @@
+# Data Layer
+
+## Redux
+
+In eHoldings, data is connected on `routes` only. This is done via
+`react-redux`'s `connect` decorator.
+
+``` javascript
+connect(({ eholdings: { data } }) => ({
+  // ...mapStateToProps
+  // data needed by the component is accessed here
+}), {
+  // ...mapDispatchToProps
+  // actions used to send requests for data are defined here
+});
+```
+
+When data is needed, you must first request it with one of the actions
+defined in `mapDispatchToProps`. The data received in
+`mapStateToProps` will then update with a pending request, and update
+again once the request resolves or rejects.
+
+## Models
+
+In eHoldings, you must define a model to request and store data in the
+redux store.
+
+``` javascript
+// src/redux/provider.js
+import model, { hasMany } from './model';
+
+class Provider {
+  // if an attribute is not defined here, the data will not be accessible in our component
+  name = '';
+  packagesSelected = 0;
+  packagesTotal = 0;
+
+  // relationships are defined using `hasMany(type)` or `belongsTo(type)`
+  // the type defaults to the relationship key
+  packages = hasMany();
+}
+
+export default model({
+  type: 'providers',           // used as the key when storing data
+  path: '/eholdings/providers' // used in the URL during requests
+})(Provider);
+```
+
+### Requesting & Resolving Data
+
+Using this model and the eHoldings resolver within `connect` allows us
```

Comment:
`eHoldings resolver` has not yet been defined.  and i dunno if i'd mention `okapi` specifically here since it's not _really_ a server, or at least not primarily.  the request goes through okapi but the data is coming from the backend module (mod-kb-ebsco, mod-go-kb, mod-et-cetera)

Reply:
I tried rewording this to be an introduction to the resolver instead of pretending we already know what it is.